test/cpp/api/functional.cpp

```diff
@@ -272,6 +272,13 @@ TEST_F(FunctionalTest, Hardshrink) {
   }
 }
 
+TEST_F(FunctionalTest, Arange) {
+  { // Test #1
+    auto x = torch::arange(0, 5);
+    TORCH_INTERNAL_ASSERT(x.dtype() == at::ScalarType::Long);
+  }
```

Comment:
@eellison I marked this PR as BC-breaking, because currently in C++ API, `torch::arange(0, 5).dtype()` prints `float`.